src/reference/antora/modules/ROOT/pages/cloudevents.adoc

```diff
@@ -0,0 +1,268 @@
+[[cloudevents]]
+= CloudEvents Support
+
+Spring Integration provides transformers (starting with version 7.0) for transforming messages into CloudEvent messages.
+It is fully based on the https://github.com/cloudevents/sdk-java[CloudEvents SDK] project.
+
+This dependency is required for the project:
+
+[tabs]
+======
+Maven::
++
+[source, xml, subs="normal", role="primary"]
+----
+<dependency>
+    <groupId>org.springframework.integration</groupId>
+    <artifactId>spring-integration-cloudevents</artifactId>
+    <version>{project-version}</version>
+</dependency>
+----
+
+Gradle::
++
+[source, groovy, subs="normal", role="secondary"]
+----
+compile "org.springframework.integration:spring-integration-cloudevents:{project-version}"
+----
+======
+
+[[cloudevent-transformers]]
+
+== CloudEvent Transformers
+
+[[tocloudeventstransformer]]
+=== ToCloudEventsTransformer
+
+The `ToCloudEventsTransformer` converts Spring Integration messages into CloudEvents compliant messages.
+This transformer provides  support for the CloudEvents specification v1.0 with configurable output format and defining attributes using `Expression` s and identifying extensions in the message headers via patterns.
+
+[[cloudevent-transformer-overview]]
+==== Overview
+
+The CloudEvents transformer (`ToCloudEventsTransformer`) converts messages to CloudEvents format.
+The CloudEvents transformer utilizes `io.cloudevents.core.provider.EventFormatProvider` to find `EventFormat` classes in the classpath and registers these as the available serializers for CloudEvents.
+The type of serialization (JSON, XML, AVRO, etc) of the CloudEvents' message is determined by `contentType` of the message.
+
+NOTE: Messages to be transformed must have a payload of `byte[]`.
+
+
+
+[[configure-transformer]]
+===== Configuring Transformer
+
+The `ToCloudEventsTransformer` allows the user to use SpEL `Expression`s to populate the attributes.
+Extensions are populated from the headers using pattern matching.
+
+====== Attribute Expressions
```

Comment:
Why do we need so many sub-titles?
Why cannot we express everything in the same section, but different paragraphs?

Reply:
No.  I'll be removing many of these.